### PR TITLE
Update datadog_operator.md: Add `site` to the sample YAML

### PR DESCRIPTION
**Pull request by scottgerring**
Adds `site` to the example YAML for the operator setup and clarify app key config

### What does this PR do? What is the motivation?
There is no mention of the `global.site` field in the example YAML. Without going through the linked CRD field spec, users who are not using the default site datadoghq.com will struggle to get the agent running. The error messages received lead down a "bad credentials" rabbit hole which does not make it immediately obvious that the site might be wrong, so I think it is a good idea to point out this needs to be set, here, on the getting started pages. 

I've also expanded on adding the app key as again, I think this is something most people will need, and it is better to be explicit about this. 

### Merge instructions

Merge queue is enabled in this repo. To have it automatically merged after it receives the required reviews, create the PR (from a branch that follows the `<yourname>/description` naming convention) and then add the following PR comment:

```
/merge
```

### CODE REVIEW

File: content/en/getting_started/containers/datadog_operator.md [hunk 1]
```diff
@@ -32,8 +32,8 @@ The [Datadog Operator][1] is an open source [Kubernetes Operator][2] that enable
   kubectl create secret generic datadog-secret --from-literal api-key=<DATADOG_API_KEY>
   ```
   Replace `<DATADOG_API_KEY>` with your [Datadog API key][5].
-
-  **Note**: add the application key for autoscaling using the external metrics server.
+  
+  **Note**: add an application key for autoscaling using the external metrics server by adding `--from-literal app-key=<DATADOG_APP_KEY>`
```
Comment (rtrieu) [inline content/en/getting_started/containers/datadog_operator.md:36]:
```suggestion
  **Note**: Add an application key for autoscaling using the external metrics server by adding `--from-literal app-key=<DATADOG_APP_KEY>`
```

File: content/en/getting_started/containers/datadog_operator.md [hunk 2]
```diff
@@ -53,6 +54,8 @@ The [Datadog Operator][1] is an open source [Kubernetes Operator][2] that enable
       logCollection:
         enabled: true
   ```
+  **Note**: Make sure to set `site` to the datadog site you are using - for instance, **datadoghq.eu**.
```
Comment (rtrieu) [inline content/en/getting_started/containers/datadog_operator.md:57]:
```suggestion
  **Note**: Make sure to set `site` to the Datadog site you are using (for instance, `datadoghq.eu`). 
```

**Review by rtrieu — CHANGES_REQUESTED** [review]
Minor feedback for style consistency.

**Comment by scottgerring** [comment]
> Minor feedback for style consistency.

Cheers @rtrieu! Applied

**Review by rtrieu — APPROVED** [review]

**Pull request merged**